aci-builder/bin-run/builder.go

```diff
@@ -123,6 +124,8 @@ func (b *Builder) writeManifest() error {
 		aciManifest.NameAndVersion = *common.NewACFullName(aciManifest.NameAndVersion.Name() + ":" + common.GenerateVersion(b.aciTargetPath))
 	}
 
+	b.fullManifest = aciManifest
```

Comment:
Instead of storing fullManifest in `Builder`, it should be return by `WriteManifest()` and passed to `Tar()`.
Here you are storing the result of the manifest processing globally in the middle of the build process. This leads to think it's always available which is not true.